server/src/main/java/org/elasticsearch/action/admin/indices/retention/TransportSyncRetentionLeasesAction.java

```diff
@@ -82,12 +81,12 @@ protected void shardOperation(SyncRetentionLeasesRequest request, ShardRouting s
         indexShard.runUnderPrimaryPermit(
             () -> indexShard.syncRetentionLeases(true, listener),
             e -> {
-                logger.warn(new ParameterizedMessage("failed to sync retention leases on OPTIMIZE TABLE"), e);
+                logger.warn("Retention lease sync failed on shard [{}]", shardRouting.shortSummary(), e);
```

Comment:
Be aware, there are no logger (warn/info/debug) methods to pass in arguments and an exception. In your code, the exception `e` is passed in as an string format argument and not as a cause.
```suggestion
                logger.warn("Retention lease sync failed on shard " + indexShard.shardId(), e);
```

Reply:
>the exception e is passed in as a string format argument

oh, right, thanks for catching!